src/protocol/sort/bit_permutations.rs

```diff
@@ -74,9 +47,28 @@ impl<'a, F: Field> BitPermutations<'a, F> {
         &self,
         ctx: &ProtocolContext<'_, N>,
     ) -> Result<Vec<Replicated<F>>, BoxError> {
-        let mult_input = self.prepare_mult_inputs(ctx);
-        let async_multiply =
-            mult_input.map(|input| async move { self.secure_multiply(ctx, input).await });
+        let share_of_one = Replicated::one(ctx.role());
+
+        let mult_input = self
+            .input
+            .iter()
+            .map(move |x: &Replicated<F>| share_of_one - *x)
+            .chain(self.input.iter().copied())
+            .enumerate()
+            .scan(Replicated::<F>::new(F::ZERO, F::ZERO), |sum, (index, x)| {
+                *sum += x;
+                Some((
+                    ctx.narrow(&format!("record_{}", index)),
```

Comment:
This looks like a mistake to me.  Any reason not to clone the context as here?

As in std::iter::repeat()

Reply:
I'm not sure exactly what you're suggesting I should replace this with.

If I swap this line with `ctx.clone()` it does not work. It tells me: 
`cannot move out of `*ctx` which is behind a shared reference`

I'm not sure how you're suggesting I use `std::iter::repeat()`, could you share a snippet of code?